File: packages/gbnf-experiment/data/20260909T023701Z_e94dca40/ported_implementation/src/grammar-graph/pointers.ts
import type { GraphPointer } from './graph-pointer.js';

export class Pointers {
  private pointers = new Map<string, GraphPointer>();

  constructor(...pointers: GraphPointer[]) {
    for (const pointer of pointers) {
      this.add(pointer);
    }
  }

  add(pointer: GraphPointer): void {
    this.pointers.set(pointer.id, pointer);
  }

  *[Symbol.iterator](): IterableIterator<GraphPointer> {
    yield* this.pointers.values();
  }

  get size(): number {
    return this.pointers.size;
  }
}

File: proposed-projects/round-trip-experiment/reverse/20260909T123239Z_5982b634/20260910T070936Z_fb630842/ported_implementation/gbnf/grammar_graph/pointers.py
from __future__ import annotations

from typing import Dict, Iterator


class Pointers:
    def __init__(self, *pointers) -> None:
        self._pointers: Dict[str, 'GraphPointer'] = {}  # noqa: F821
        for pointer in pointers:
            self.add(pointer)

    def add(self, pointer) -> None:
        self._pointers[pointer.id] = pointer

    def __iter__(self) -> Iterator:
        return iter(list(self._pointers.values()))

    @property
    def size(self) -> int:
        return len(self._pointers)
